extensions/wikia/Email/Controller/FounderDigestController.class.php

```diff
@@ -0,0 +1,310 @@
+				],
+				[
+					'type' => 'text',
+					'name' => 'pageViews',
+					'label' => 'Page views',
+					'tooltip' => 'Number of page views'
+				]
+			]
+		];
+
+		return array_merge_recursive( $formFields, parent::getEmailSpecificFormFields() );
+	}
+}
+
+class FounderActivityDigestController extends FounderDigestController {
+	const TRACKING_CATEGORY_EN = TrackingCategories::FOUNDER_ACTIVITY_DIGEST_EN;
+	const TRACKING_CATEGORY_INT = TrackingCategories::FOUNDER_ACTIVITY_DIGEST_INT;
+
+	protected $pageEdits;
+	protected $newUsers;
+
+	public function initEmail() {
+		$this->pageEdits = $this->request->getVal( 'pageEdits' );
+		$this->newUsers = $this->request->getVal( 'newUsers' );
+		// Parent method is called here so that assertion of parameters is done at the right time
+		parent::initEmail();
+		$this->pageEdits = number_format( $this->pageEdits );
+		$this->newUsers = number_format( $this->newUsers );
+	}
+
+	protected function getSubject() {
+		return $this->getMessage( 'emailext-founder-activity-digest-subject',
+			\WikiMap::getWikiName( $this->wikiId ) )->parse();
+	}
+
+	/**
+	 * @template digestLayout
+	 */
+	public function body() {
+		$this->response->setData( [
+			'salutation' => $this->getSalutation(),
+			'summary' => $this->getSummary(),
+			'buttonText' => $this->getButtonText(),
+			'buttonLink' => $this->getButtonLink(),
+			'details' => $this->getDetailsList(),
+			'contentFooterMessages' => [
+				$this->getCommunityFooterMessage()
+			],
+			'hasContentFooterMessages' => true
+		] );
+	}
+
+	public function assertCanEmail() {
+		parent::assertCanEmail();
+		$this->assertSubscribedToCompleteDigest();
+	}
+
+	/**
+	 * Asserts that the founder is subscribed to the complete digest.
+	 *
+	 * @throws \Email\Check
+	 */
+	protected function assertSubscribedToCompleteDigest() {
+		if ( !$this->targetUser->getBoolOption( 'founderemails-complete-digest-' . $this->wikiId ) ) {
+			throw new Check( 'Founder is not subscribed to complete digest.' );
+		}
+	}
+
+	protected function assertValidParams() {
+		parent::assertValidParams();
+		$this->assertNewUsersSet();
+		$this->assertPageEditsSet();
+	}
+
+	/**
+	 * Asserts that the `newUsers` value is passed
+	 *
+	 * @throws \Email\Check
+	 */
+	protected function assertNewUsersSet() {
+		if ( empty( $this->newUsers ) && $this->newUsers !== '0' ) {
+			throw new Check( 'Invalid value passed for `newUsers`' );
+		}
+	}
+
+	/**
+	 * Asserts that the `pageEdits` value is passed
+	 *
+	 * @throws \Email\Check
+	 */
+	protected function assertPageEditsSet() {
+		if ( empty( $this->pageEdits ) && $this->pageEdits !== '0' ) {
+			throw new Check( 'Invalid value passed for `pageEdits`' );
+		}
+	}
+
+	/**
+	 * Returns list of details for the digest
+	 *
+	 * @return array
+	 */
+	protected function getDetailsList() {
+		return [
+			[
+				'iconSrc' => \GlobalFile::newFromText( 'Page-views.png', \Wikia::NEWSLETTER_WIKI_ID )->getUrlGenerator()->url(),
```

Comment:
OK, I noticed EmailImageHelper only supports GIF files. So I'll have to change it to accept other image types.